### ISSUE TITLE
Expand lru_crawler metadump all to allow basic filter on keys matched/found.

### ISSUE THREAD

**Issue by pppchris** [memcached/memcached#759]
Really an idea, the addition of the lru_crawler has been an amazing addition. An idea to expand on the current functionality to allow a basic filter when returning the list. 

Using most clients it isnt possible to grep/filter and so the response size is huge. Having the ability to include a basic filter such key starts with would at least minimize that and allow more possibilities.

**Comment by dormando** [comment]
Hey,

I initially didn't do this because I don't want people to rely on it inline; even with filtering you can still only run one search at a time, and it's going to take a similar amount of time to run as if you filter it with grep since internally we still examine every key.

that said it'd be nice to have. This and a few other recent requests makes me regret how the command syntax was done (we now have hash based + lru based walking, the system is pretty modular internally, etc). So I need to give this a re-think and maybe a new command.

**Comment by pppchris** [comment]
Ah that would explain it as weve used it inline. Its surprising how many use cases can appear when you can see all keys. Its really powerful. 

In this use case we are using lru_crawler to find some specific items that we need to purge from a big set but dont nessessarily know the full key (just the start).. Either way great work. Thanks for considering.

**Comment by dormando** [comment]
Yeah.. to be clear it's a matter of scalability. Many folks have instances that can be searched in a quarter second; tens of thousands to hundreds of thousands of keys. Many others have billions and can take several minutes. The feature is designed so it can't hang or slow down the server but anyone using the command inline with their app is in for trouble if they rely on it and then scale. So I was making it awkward to use deliberately.

**Comment by pppchris** [comment]
Ah. I can see a few people falling into the trap. Its really really powerful. 

Other than flush all there isnt really way to do a bulk purge unless you have the keys?

**Comment by dormando** [comment]
People do it a lot via namespaces (ie; indirect keys). Add an extra lookup for the ability to expire all related keys at the same time.
